core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -787,9 +762,39 @@ private static ConfigResponse fetchConfig(
   private AuthSession newSession(
       Map<String, String> credentials, Map<String, String> properties, AuthSession parent) {
     if (credentials != null) {
-      // use the bearer token without exchanging
       if (credentials.containsKey(OAuth2Properties.TOKEN)) {
-        return newSession(credentials.get(OAuth2Properties.TOKEN), expiresInMs(properties), parent);
+        String token = credentials.get(OAuth2Properties.TOKEN);
+        Optional<JWT> jwt = JWT.of(token);
+
+        if (OAuth2Util.tokenExpired(jwt)) {
```

Comment:
The logic for token expiration should be handled in `sessionFromToken` and other OAuth2Util methods.